ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/precompiles/privacy/PrivacyPrecompiledContract.java

```diff
@@ -80,31 +80,32 @@ public void setPrivateTransactionProcessor(
   }
 
   @Override
-  public Gas gasRequirement(final BytesValue input) {
+  public Gas gasRequirement(final Bytes input) {
     return Gas.of(40_000L); // Not sure
   }
 
   @Override
-  public BytesValue compute(final BytesValue input, final MessageFrame messageFrame) {
-    final String key = BytesValues.asBase64String(input);
+  public Bytes compute(final Bytes input, final MessageFrame messageFrame) {
+    final String key = input.toBase64String();
     final ReceiveRequest receiveRequest = new ReceiveRequest(key);
 
     final ReceiveResponse receiveResponse;
     try {
       receiveResponse = enclave.receive(receiveRequest);
     } catch (final Exception e) {
       LOG.error("Enclave probably does not have private transaction with key {}.", key, e);
-      return BytesValue.EMPTY;
+      return Bytes.EMPTY;
     }
 
     final BytesValueRLPInput bytesValueRLPInput =
-        new BytesValueRLPInput(BytesValues.fromBase64(receiveResponse.getPayload()), false);
+        new BytesValueRLPInput(
```

Comment:
(`java.util.`)`Base64.getDecoder().decode(bytes)`?  It saves us a String instantiation.

Reply:
ok done.